### PR TITLE
Jack issues 2 calendar button

### PR DESCRIPTION
**Pull request by jHaben**
@mftruso  it seems to be working. when you have some free time I would like to go over with you. thanks.

### CODE REVIEW

File: src/utils/vevent.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export default class Vevent {
+
+    constructor(start, end) {
+
+       this.productId = '-//St. Isidore Guild//Ember Days App//EN';
+       this.title = 'Ember Day';
+       this.description = "Prayer and Fasting";
+       this.start = start;
+       this.end = end;
```
Comment (mftruso) [inline src/utils/vevent.js:11]: can calculating an end date be replaced with `this.duration = { days: 1 }`?

**Review by mftruso — COMMENTED** [review]
Nice work! Just a few suggestions to simplify your code.

**Comment by mftruso** [comment]
also, be sure to rebase on master so the checks pass

**Comment by jHaben** [comment]
thanks Mike! I will make the changes

File: src/utils/icsFileUtils.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import Vevent from '../utils/vevent';
+
+export function createIcsFile(eventDates) {
+
+    const ics = require('ics');
```
[the remark this answers is not in the document]
Comment (jHaben) [reply src/utils/icsFileUtils.js:5]: done.

File: src/components/CalendarGenerator.vue [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+} from "../utils/dateUtils";
+import { calculateAllDates } from "@/utils/dateUtils";
+
+export default {
+  name: "CalendarGenerator",
+  props: { year: Number },
+  methods: {
+    addToCalendar: function () {
+      let dates = [
+        calculateAllDates(calculateFall(this.year)),
```
[the remark this answers is not in the document]
Comment (jHaben) [reply src/components/CalendarGenerator.vue:28]: This was good to learn about. Overthought it at first but I think I got it.

File: src/components/CalendarGenerator.vue [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  calculateSummer,
+} from "../utils/dateUtils";
+import { calculateAllDates } from "@/utils/dateUtils";
+
+export default {
+  name: "CalendarGenerator",
+  props: { year: Number },
+  methods: {
+    addToCalendar: function () {
+      let dates = [
```
[the remark this answers is not in the document]
Comment (jHaben) [reply src/components/CalendarGenerator.vue:27]: done.

File: src/utils/icsFileUtils.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    document.body.appendChild(elem);
+    elem.click();
+    document.body.removeChild(elem);
+
+}
+
+export function createIcsEventDates(dates) {
+    const events = []; //Array of ember days with ics attributes to be fed into ics file. 
+
+    for (let i = 0; i < dates.length; i++) {
```
[the remark this answers is not in the document]
Comment (jHaben) [reply src/utils/icsFileUtils.js:35]: done.

File: src/utils/vevent.js (hunk 1 above)
Comment (jHaben) [reply src/utils/vevent.js:11]: good catch. much easier. done.

**Comment by mftruso** [comment]
Looks good! I just added a commit making the vuex getters a little more reactive. Thanks for all your help on this issue @jHaben!

**Pull request merged**